src/request.ts

```diff
@@ -203,3 +203,17 @@ filterRegistry.register(
 		};
 	}
 );
+
+/**
+ * Add a filter that automatically parses incoming Buffer responses in Node.
+ */
+filterRegistry.register(
+	function (response: Response<any>, url: string, options: RequestOptions) {
+		return Buffer.isBuffer(response.data) !== undefined && options.responseType === 'json';
```

Comment:
I am afraid you are accessing Buffer in an unsafe way, which is [throwing errors](https://travis-ci.org/dojo/core/builds/95615947#L388) in browser environments.  Also, we should consider narrowest to widest in our logic here to take advantage of short cutting.

I think a better way would be something like this:

```typescript
return options.responseType === 'json' && Buffer && typeof Buffer.isBuffer(response.data) !== 'undefined';
```